src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,109 @@
+import { AuthIdentity } from "decentraland-crypto-fetch"
+import { DCLNotification, NotificationActiveTab } from "decentraland-ui/dist/components/Notifications/types"
+import { NOTIFICATIONS_QUERY_INTERVAL } from "../containers/Navbar/constants"
+import { CURRENT_AVAILABLE_NOTIFICATIONS } from "decentraland-ui/dist/components/Notifications/utils"
+
+const useNotifications = (identity: AuthIdentity | undefined, isNotificationsEnabled: boolean) => {
+  const [{ isLoading, notifications }, setUserNotifications] = useState<{
+    isLoading: boolean
+    notifications: DCLNotification[]
+  }>({
+    isLoading: false,
+    notifications: []
+  })
+
+  const [{ activeTab, isOnboarding, isOpen }, setNotificationsState] = useState({
+    activeTab: NotificationActiveTab.NEWEST,
+    isOnboarding: checkIsOnboarding(),
+    isOpen: false
+  })
+
+  const notificationsClient: NotificationsAPI | null = useMemo(() => {
+    if (identity) return new NotificationsAPI({ identity })
+    return null
+  }, [identity])
+
+  const handleOnBegin = () => {
+    setOnboardingDone()
+    setNotificationsState(prevState => ({ ...prevState, isOnboarding: false }))
+  }
+
+  const handleNotificationsOpen = async () => {
+    const currentOpenState = isOpen
+
+    setNotificationsState(prevState => {
+      return { ...prevState, isOpen: !prevState.isOpen }
+    })
+
+    if (!currentOpenState) {
+      const unreadNotifications = notifications
+        .filter(notification => !notification.read)
+        .map(({ id }) => id)
+      if (unreadNotifications.length) {
+        await notificationsClient?.markNotificationsAsRead(unreadNotifications)
+      }
+    } else {
+      // update state when closes the modal
+      const markNotificationAsReadInState = notifications.map(notification => {
+        if (notification.read) return notification
+
+        return {
+          ...notification,
+          read: true
+        }
+      })
+      setUserNotifications({
+        isLoading,
+        notifications: markNotificationAsReadInState
+      })
+    }
+  }
+
+  const handleOnChangeModalTab = (tab: NotificationActiveTab) => setNotificationsState(prevState => ({
+    ...prevState,
+    activeTab: tab
+  }))
+
+  const fetchNotificationsState = () => {
+    setUserNotifications({ notifications: [], isLoading: true })
+    notificationsClient?.getNotifications().then(retrievedNotifications => {
+      setUserNotifications({
+        isLoading: false,
+        notifications: retrievedNotifications.filter(notification => CURRENT_AVAILABLE_NOTIFICATIONS.includes(notification.type))
+      })
+    })
+  }
+
+  useEffect(() => {
+    if (identity && isNotificationsEnabled) {
+      fetchNotificationsState()
+
+      const interval = setInterval(() => {
+        fetchNotificationsState()
+      }, NOTIFICATIONS_QUERY_INTERVAL)
+
+      return () => {
+        clearInterval(interval)
+      }
+    } else {
+      return () => {}
+    }
```

Comment:
do we need this? in case we do, I'll do it like this

```suggestion
    }
    return () => {}

```